src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lowerBase;
+    private double upperBase;
+    private double side;
+
+    public IsoscelesTrapezoid(Color color, double lowerBase, double upperBase, double side) {
+        super(color);
+        this.lowerBase = lowerBase;
+        this.upperBase = upperBase;
+        this.side = side;
+    }
+
+    @Override
+    public double calculateArea() {
+        return ((lowerBase + upperBase) / 2) * (Math.sqrt(Math.pow(side, 2)
+                - Math.pow((lowerBase - upperBase), 2) / 4));
+    }
+
+    @Override
+    public void drawFigure() {
```

Comment:
The method name 'drawFigure' could be improved to be more consistent with the class naming conventions. Consider renaming it to 'draw' to align with the task requirements of having single-purpose methods and classes.